Tasks/AzureVmssDeployment/operations/VirtualMachineScaleSet.ts

```diff
@@ -129,54 +129,87 @@ export default class VirtualMachineScaleSet {
         try {
             customScriptInfo.storageAccount = await this._getStorageAccountDetails();
             customScriptInfo.blobUris = await this._uploadCustomScriptsToBlobService(customScriptInfo);
-        } catch(error) {
+        } catch (error) {
             throw tl.loc("UploadingToStorageBlobsFailed", error.message ? error.message : error);
         }
 
         return customScriptInfo;
     }
 
     private _archiveCustomScripts(osType: string): CustomScriptsInfo {
-        try {
-            console.log(tl.loc("ArchivingCustomScripts", this.taskParameters.customScriptsPath));
-            let archive: ArchiveInfo = this._computeArchiveDetails(osType);
+        let escapedScript;
+        let escapedArgs;
+        let archiveFile = "";
 
-            if(!tl.exist(archive.directory)) {
-                tl.mkdirP(archive.directory);
-            }
+        let packageDirectory = this.taskParameters.customScriptsDirectory;
 
-            // create archive file
-            compress.createArchive(this.taskParameters.customScriptsPath, archive.compression, archive.filePath);
-
-            let invokerScriptPath: string;
-            let invokerCommand: string;
-            let escapedUserCommand = this.taskParameters.customScriptCommand.replace(/'/g, "''");
-            if(osType === "Windows") {
-                invokerScriptPath = path.join(__dirname, "..", "Resources", "customScriptInvoker.ps1");
-                invokerCommand = `powershell ./customScriptInvoker.ps1 -zipName '${archive.fileName}' -command '${escapedUserCommand}'`;
-            } else {
-                invokerScriptPath = path.join(__dirname, "..", "Resources", "customScriptInvoker.sh");
-                invokerCommand = `./customScriptInvoker.sh '${archive.fileName}' '${escapedUserCommand}'`;
-            }
+        if (osType === "Windows") {
+            // escape powershell special characters. This is needed as this script will be executed in a powershell session
+            let script = this.taskParameters.customScript.replace(/`/g, '``').replace(/\$/g, '`$');
 
-            // copy invoker script to same dir as archive
-            tl.cp(invokerScriptPath, archive.directory, "-f", false);
-            console.log(tl.loc("CopiedInvokerScript", archive.directory));
+            // put an extra quote to handle space in script name
+            let quotedScript = `.\\\\"${script}\"`
```

Comment:
what are the four slashes for. later for escaping quote you are using single slash

Reply:
there is an extra slash to start filename with .\ so that powershell does not crib that file not found